shared/reducers/signup.js

```diff
@@ -0,0 +1,30 @@
+/* @flow */
+
+import * as Constants from '../constants/signup'
+
+import type {SignupActions, CheckInviteCode} from '../constants/signup'
+
+export type SignupState = {
+  inviteCodeError: ?string,
+}
+
+const initialState: SignupState = {
+  inviteCodeError: null
+}
+
+export default function (state: SignupState = initialState, action: SignupActions): SignupState {
+  switch (action.type) {
+    case Constants.checkInviteCode:
+      // Assert this is the type we're looking at
+      var _t: CheckInviteCode = action
```

Comment:
If we're doing this to be static typed we should just use this type below and not the raw action